### ISSUE TITLE
xUnit console runner crashes when writing invalid characters to the XML file

### ISSUE THREAD

**Issue by jamesqo** [dotnet/corefx#7166]
Separating this out from #7042 (where the bug was discovered) so we have a separate issue to keep track of it.

Currently, the xUnit console runner will crash if certain invalid characters, like standalone surrogates, U+FFFE or U+FFFF are written to the XML file. For example, this would fail:

```csharp
public static IEnumerable<object[]> Equal_TestData()
{
    yield return new object[] { "\uD800", "\uD800" };
}

[Theory]
[MemberData(nameof(Equal_TestData))]
public void Equal(string expected, string actual)
{
    Assert.Equal(expected, actual);
}
```

The fix is to alter [this](https://github.com/dotnet/buildtools/blob/37d1d57167d2d112b26f94135f3281fe2601ab65/src/common/XmlTestExecutionVisitor.cs#L269) method of `XmlTestExecutionVisitor` in the buildtools repo to handle these invalid chars, in addition to C0 controls. See the fix I've made upstream here: xunit/xunit#804

I'll be submitting a PR shortly to buildtools to fix this.

**Comment by joshfree** [comment]
@mellinoe can you assist?

**Comment by mellinoe** [comment]
@jamesqo I see that you helped with some fixes upstream in xUnit related to this. Are we able to pull those changes into our fork now?

**Comment by jamesqo** [comment]
Unfortunately no, the PR that fixes this once and for all upstream is still open: xunit/xunit#807

**Comment by mellinoe** [comment]
Ah okay. Thanks for the update. I am going to move this to Future as this is only a test issue and is blocked by an external dependency for now.

**Comment by karelz** [comment]
@jamesqo the xunit PR was merged in June. What is there left in CoreFX? Or is it purely buildtools issue? (we should move it there in such case ...)
